### PR TITLE
Pass along container to created state factory

### PR DESCRIPTION
**Pull request by jayphelps**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,12 +23,7 @@ import Ember from 'ember';
 import StateMixin from 'ember-state-services/mixin'
 
 export default Ember.Object.extend(StateMixin, {
-  stateName: 'emailEdit',
-  setupState: function(Factory, model) {
```
Comment (jayphelps) [inline README.md]: I removed because I'm confused why we would them how to re-implement the default behavior?

Comment (thoov) [inline README.md]: This sentence is kinda confusing to me :-)

Comment (jayphelps) [inline README.md]: hahaha indeed. So our readme example should show you how to use it, right? `stateName` is required, but `setupState` has a [default implementation](https://github.com/stefanpenner/ember-state-services/blob/master/addon/mixin.js#L20) so why imply they need to reimplement exactly that? They may need to override it to do some sort of other behavior on top of the default, sure, but this example does not show that. An example showing that probably should call `_super()` to get the instance (keeping container reference) then do whatever else needs to be done.

**Comment by jayphelps** [comment]
This PR isn't even needed in regards to container passing........the test snuck into my previous PR and is passing....soooooo apparently `const StateFactory = this.container.lookupFactory(stateName);` somehow auto sets the container to instances of that factory as well. I independently tested and confirmed this. Very clever of it. We can talk about the README `setupState` example still...but this PR's primary point is moot.

File: README.md (hunk 1 above)
Comment (thoov) [inline README.md]: I agree. This example isn't really demonstrating anything useful (and potentially adding confusion to a newcomer). We might want to add another section with a better example with how one might use setupState but that is probably more of an advanced use case.

Comment (jayphelps) [inline README.md]: Agreed. I'm going to just remove `setupState` in the example for now and then try and come up with some examples of overriding behavior with `_super()` calls.

**Comment by stefanpenner** [comment]
> soooooo apparently const StateFactory = this.container.lookupFactory(stateName); somehow auto sets the container to instances of that factory as well.

yes, lookupFactory builds a special factory that has the container + injections prepared.

**Pull request closed without merging**